src/main/java/com/synopsys/integration/jenkins/detect/service/strategy/DetectAirGapJarStrategy.java

```diff
@@ -0,0 +1,131 @@
+        private final Map<String, String> environmentVariables;
+        private final String remoteJdkHome;
+        private final JenkinsConfigService jenkinsConfigService;
+        private final AirGapDownloadStrategy airGapDownloadStrategy;
+
+        public SetupCallableImpl(JenkinsIntLogger logger, Map<String, String> environmentVariables, String remoteJdkHome, JenkinsConfigService jenkinsConfigService, AirGapDownloadStrategy airGapDownloadStrategy) {
+            this.logger = logger;
+            this.environmentVariables = environmentVariables;
+            this.remoteJdkHome = remoteJdkHome;
+            this.jenkinsConfigService = jenkinsConfigService;
+            this.airGapDownloadStrategy = airGapDownloadStrategy;
+        }
+
+        @Override
+        public ArrayList<String> call() throws IntegrationException {
+            String airGapJarPath = getOrDownloadAirGapJar();
+            RemoteJavaService remoteJavaService = new RemoteJavaService(logger, remoteJdkHome, environmentVariables);
+            String javaExecutablePath = remoteJavaService.calculateJavaExecutablePath();
+            remoteJavaService.logDebugData(javaExecutablePath);
+
+            logger.info("Detect configured: " + airGapJarPath);
+
+            return new ArrayList<>(Arrays.asList(javaExecutablePath, "-jar", airGapJarPath));
+        }
+
+        private String getOrDownloadAirGapJar() throws IntegrationException {
+            DetectAirGapInstallation airGapInstallation;
+            try {
+                airGapInstallation = jenkinsConfigService.getInstallationForNodeAndEnvironment(DetectAirGapInstallation.DescriptorImpl.class, airGapDownloadStrategy.getAirGapInstallationName()).get();
+            } catch (Exception e) {
+                throw new DetectJenkinsException(
+                    String.format("Problem encountered getting Detect Air Gap tool with the name %s from global tool configuration. Check your plugin and tool configuration.", airGapDownloadStrategy.getAirGapInstallationName()), e);
+            }
+
+            String airGapBaseDir = airGapInstallation.getHome();
+
+            if (airGapBaseDir == null) {
+                throw new DetectJenkinsException("Detect AirGap installation directory is null. Check your plugin and tool configuration.");
+            }
+
+            return getAirGapJar(airGapBaseDir);
+        }
+
+        private String getAirGapJar(String airGapBaseDir) throws IntegrationException {
+            FileFilter fileFilter = file -> file.getName().startsWith(DETECT_JAR_PREFIX) && file.getName().endsWith(DETECT_JAR_SUFFIX);
+            File[] foundAirGapJars = new File(airGapBaseDir).listFiles(fileFilter);
+
+            if (foundAirGapJars == null || foundAirGapJars.length == 0) {
+                throw new DetectJenkinsException(String.format("No Detect AirGap jar was found in installation directory %s. Check your plugin and tool configuration.", airGapBaseDir));
```

Comment:
I think we should be clear on what we expect here-- we expect one jar at the root directory because we expect that they are installing an unmodified air gap zip that we distributed. We should make sure they verify that they are installing that, here, in the null case above, and in the multiple jar case below.